### DISCUSSION TITLE
How to use aggregation

### DISCUSSION THREAD

**Discussion by Edvinaz** [jenssegers/laravel-mongodb#2542] [Q&A]
Sorry if I missed answer somewhere, but I cant solve one issue...

I build query from many filters and I need to get count of documents distincted by two columns. Distinct and group by seems not working... With raw aggregation I get what i need, but question is how to build aggregation if my $match stage is build with `Builder`

**Comment by GromNaN** [comment]
You can't mix the query builder and raw aggregation. An aggregation builder will be created to solve this issue.

The current way to use custom aggregation is like this:
```php
use MongoDB\Collection;

$builder->raw(function (Collection $collection) {
    return $collection->aggregate($pipeline, $options);
});
```
